hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java

```diff
@@ -204,38 +204,51 @@ public void registerListener(Listener listener1) {
   public FSDataInputStream open(final Path path, final int bufferSize) throws IOException {
     LOG.debug("AzureBlobFileSystem.open path: {} bufferSize: {}", path, bufferSize);
     // bufferSize is unused.
-    return open(path, Optional.empty());
+    return open(path, new OpenFileParameters());
```

Comment:
Concern on using this way and passing OpenFileParameters  param.
When FS#openFile() API is been used, you can see how system create the OpenFileParameters(). There may be some stuff coming in as default there. 
eg:  bufferSize = fileSystem.getConf().getInt(IO_FILE_BUFFER_SIZE_KEY,
        IO_FILE_BUFFER_SIZE_DEFAULT);

So correct buffer size as per conf will be there in OpenFileParameters also
We might not be using buffer size from the param now.  But later its possible to use some new stuff like this.
So better dont create OpenFileParameters () here and pass.
Instead we can have the private method accept Optional<OpenFileParameters>.
In open(Path), we can Optional.empty() anyways.